app/models/irregular_income_payment.rb

```diff
@@ -0,0 +1,23 @@
+class IrregularIncomePayment < ApplicationRecord
+  ANNUAL_FREQUENCY = 'annual'.freeze
+  STUDENT_LOAN = 'student_loan'.freeze
+  VALID_IRREGULAR_INCOME_FREQUENCY = [ANNUAL_FREQUENCY].freeze
+  VALID_IRREGULAR_INCOME_TYPES = [STUDENT_LOAN].freeze
+
+  belongs_to :irregular_income
+
+  validates_uniqueness_of :income_type
+  validates :amount, numericality: { greater_than_or_equal_to: 0 }
+
+  def student_loan?
+    return false unless annual?
+
```

Comment:
Do we need this here?  You're asking if it's a student loan, and responding false even if the income type is student loan but the frequency is not annual.  I'm not sure it's needed in any case - what is needed is validation on the income type and frequency.